DataStructures/Graphs/MaxFlowDinic.java

```diff
@@ -0,0 +1,160 @@
+import java.util.*;
+class MaxFlowDinic
+{
+    private int V,level[];
+    private ArrayList<Edge> adj[];
+    class Edge
```

Comment:
In latest version of Java (java 8) you don't need inner classes. You can write this sperated in one single file like:  
```java
// one file
class A {

}

class B {

}

// ... 
```